model/roles_test.go

```diff
@@ -323,3 +323,160 @@ func TestGetRoleManifestDevPackageVersion(t *testing.T) {
+
+	workDir, err := os.Getwd()
+	assert.NoError(err)
+
+	torReleasePath := filepath.Join(workDir, "../test-assets/tor-boshrelease")
+	torReleasePathBoshCache := filepath.Join(torReleasePath, "bosh-cache")
+	release, err := NewDevRelease(torReleasePath, "", "", torReleasePathBoshCache)
+	assert.NoError(err)
+
+	roleManifestPath := filepath.Join(workDir, "../test-assets/role-manifests/variables-without-decl.yml")
+	rolesManifest, err := LoadRoleManifest(roleManifestPath, []*Release{release})
+	assert.Equal(err.Error(),
+		`configuration.templates: Not found: "No variable declaration of 'HOME'"`)
+	assert.Nil(rolesManifest)
+}
+
+func TestLoadRoleManifestNonTemplates(t *testing.T) {
+	assert := assert.New(t)
+
+	workDir, err := os.Getwd()
+	assert.NoError(err)
+
+	torReleasePath := filepath.Join(workDir, "../test-assets/tor-boshrelease")
+	torReleasePathBoshCache := filepath.Join(torReleasePath, "bosh-cache")
+	release, err := NewDevRelease(torReleasePath, "", "", torReleasePathBoshCache)
+	assert.NoError(err)
+
+	roleManifestPath := filepath.Join(workDir, "../test-assets/role-manifests/templates-non.yml")
+	rolesManifest, err := LoadRoleManifest(roleManifestPath, []*Release{release})
+	assert.Equal(err.Error(),
+		`configuration.templates: Invalid value: "": Using 'properties.tor.hostname' as a constant`)
+	assert.Nil(rolesManifest)
+}
+
+func TestLoadRoleManifestRunEnvDocker(t *testing.T) {
+	assert := assert.New(t)
+
+	workDir, err := os.Getwd()
+	assert.NoError(err)
+
+	torReleasePath := filepath.Join(workDir, "../test-assets/tor-boshrelease")
+	torReleasePathBoshCache := filepath.Join(torReleasePath, "bosh-cache")
+	release, err := NewDevRelease(torReleasePath, "", "", torReleasePathBoshCache)
+	assert.NoError(err)
+
+	roleManifestPath := filepath.Join(workDir, "../test-assets/role-manifests/docker-run-env.yml")
+	rolesManifest, err := LoadRoleManifest(roleManifestPath, []*Release{release})
+	assert.Equal(err.Error(),
+		`Role 'dockerrole': run.env: Not found: "No variable declaration of 'UNKNOWN'"`)
+	assert.Nil(rolesManifest)
+}
+
+func TestLoadRoleManifestRunGeneral(t *testing.T) {
+	assert := assert.New(t)
+
+	workDir, err := os.Getwd()
+	assert.NoError(err)
+
+	torReleasePath := filepath.Join(workDir, "../test-assets/tor-boshrelease")
+	torReleasePathBoshCache := filepath.Join(torReleasePath, "bosh-cache")
+	release, err := NewDevRelease(torReleasePath, "", "", torReleasePathBoshCache)
+	assert.NoError(err)
+
+	tests := []struct {
+		manifest string
+		message  string
+	}{
+		{
+			"bosh-run-missing.yml",
+			`Role 'myrole': run: Required value`,
+		},
+		{
+			"bosh-run-bad-proto.yml",
+			`Role 'myrole': run.exposed-ports[https].protocol: Unsupported value: "AA": supported values: TCP, UDP`,
+		},
+		{
+			"bosh-run-bad-ports.yml",
+			`Role 'myrole': run.exposed-ports[https].external: Invalid value: 0: must be between 1 and 65535, inclusive
+Role 'myrole': run.exposed-ports[https].internal: Invalid value: "-1": invalid syntax`,
```

Comment:
Should be, at least in the definition of the expected messages. No indentation stuff to think about.